src/app/modules/item/pages/item-forum/item-forum.component.html

```diff
@@ -34,56 +34,54 @@ <h2 class="title alg-h2">Help requests</h2>
+              <th></th>
+              <th *ngIf="itemData?.item?.type !== 'Task'" i18n>Content</th>
+              <th i18n>User</th>
+              <th i18n>Status</th>
+              <th i18n># msgs</th>
+              <th i18n>Latest update</th>
             </tr>
           </ng-template>
 
           <ng-template
             pTemplate="body"
             let-rowData
-            let-columns="columns"
           >
-            <tr>
-              <td *ngFor="let col of columns">
-                <ng-container [ngSwitch]="col.field">
-                  <ng-container *ngSwitchCase="'item.title'">
-                    <a class="alg-link" [ngClass]="{'disabled': !rowData.item}" [routerLink]="rowData.item | rawItemRoute | url">
-                      {{ rowData.item.title }}
-                    </a>
-                  </ng-container>
-                  <ng-container *ngSwitchCase="'participant'">
-                    <a class="alg-link" [routerLink]="{ id: rowData.participant.id, isUser: true } | groupLink">
-                      {{ rowData.participant | userCaption }}
-                    </a>
-                  </ng-container>
-                  <ng-container *ngSwitchCase="'status'">
-                    {{ rowData.status | threadStatusDisplay }}
-                  </ng-container>
-                  <ng-container *ngSwitchCase="'latestUpdateAt'">
-                    {{ rowData.latestUpdateAt | date:'short' }}
-                  </ng-container>
-                  <ng-container *ngSwitchDefault>
-                    {{ rowData[col.field] }}
-                  </ng-container>
-                </ng-container>
+            <tr *ngrxLet="{
+                id: { participantId: rowData.participant.id, itemId: rowData.item.id },
+                isVisible: discussion.isVisible && rowData.participant.id === discussion.threadId?.participantId && rowData.item.id === discussion.threadId?.itemId,
+              } as rowThread"
+            >
+              <td>
+                <button pButton type="button" class="p-button-rounded" *ngIf="!rowThread.isVisible" (click)="toggleVisibility(true, rowThread.id)" i18n>Open</button>
+                <button pButton type="button" class="p-button-rounded p-button-danger" *ngIf="rowThread.isVisible" (click)="toggleVisibility(false)" i18n>Close</button>
+              </td>
+              <td *ngIf="itemData?.item?.type !== 'Task'">
+                <a class="alg-link" [ngClass]="{'disabled': !rowData.item}" [routerLink]="rowData.item | rawItemRoute | url">
+                  {{ rowData.item.title }}
+                </a>
+              </td>
+              <td>
+                <a class="alg-link" [routerLink]="{ id: rowData.participant.id, isUser: true } | groupLink">
+                  {{ rowData.participant | userCaption }}
+                </a>
               </td>
+              <td>{{ rowData.status | threadStatusDisplay }}</td>
+              <td>{{ rowData['messageCount'] }}</td>
```

Comment:
Didn't get why not dot notation

Reply:
Good question. That's a paste from Mathieu's work to keep only some parts... I think it comes from the previous version `rowData[col.field]` ... but I will change it for consistency.

Reply:
Got it